src/orbits/keplerian.jl

```diff
@@ -0,0 +1,532 @@
+struct KeplerianOrbit{T,L,D,R,A,I,M} <: AbstractOrbit
+    period::T
+    t_0::T
+    t_p::T
+    t_ref::T
+    duration::Union{Nothing, T}
+    a::L
+    a_planet::L
+    a_star::L
+    R_planet::Union{Nothing, L}
+    R_star::L
+    rho_planet::Union{Nothing, D}
+    rho_star::D
+    RpRs::R
+    aR_star::R
+    b::R
+    ecc::R
+    M_0::R
+    cos_incl::R
+    sin_incl::R
+    cos_omega::R
+    sin_omega::R
+    cos_Omega::R
+    sin_Omega::R
+    incl::A
+    omega::A
+    Omega::A
+    n::I
+    M_planet::M
+    M_star::M
+end
+
+function normalize_inputs(
+    period, t_0, t_p, t_ref, duration,
+    a, a_planet, a_star, R_planet, R_star,
+    rho_planet, rho_star,
+    RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega,
+    incl, omega, Omega,
+    M_planet, M_star,
+    no_units,
+    )
+
+    # Normalize dimensionless quantities
+    RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega = promote(
+        RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega
+    )
+
+    # Normalize remaining quantities
+    if no_units
+        period, t_0, t_p, t_ref = promote(period, t_0, t_p, t_ref)
+        !isnothing(duration) && (duration = convert(typeof(period), duration))
+        a, a_planet, a_star, R_star = promote(a, a_planet, a_star, R_star)
+        !isnothing(R_planet) && (R_planet = convert(typeof(a), R_planet))
+        !isnothing(rho_planet) && (rho_planet = convert(typeof(rho_star), rho_planet))
+        incl, omega, Omega = promote(incl, omega, Omega)
+        M_planet, M_star = promote(M_planet, M_star)
+    else
+        period, t_0, t_p, t_ref = uconvert.(u"d", (period, t_0, t_p, t_ref))
+        !isnothing(duration) && (duration = uconvert(u"d", duration))
+        a, a_planet, a_star, R_star = uconvert.(u"Rsun", (a, a_planet, a_star, R_star))
+        !isnothing(R_planet) && (R_planet = uconvert(u"Rsun", R_planet))
+        rho_star = uconvert(u"Msun/Rsun^3", rho_star)
+        !isnothing(rho_planet) && (rho_planet = uconvert(u"Msun/Rsun^3", rho_planet))
+        incl, omega, Omega = uconvert.(u"rad", (incl, omega, Omega))
+        M_planet, M_star = uconvert.(u"Msun", (M_planet, M_star))
+    end
+
+    return (
+        period, t_0, t_p, t_ref, duration,
+        a, a_planet, a_star, R_planet, R_star,
+        rho_planet, rho_star,
+        RpRs, aR_star, b, ecc, M_0, cos_incl, sin_incl, cos_omega, sin_omega, cos_Omega, sin_Omega,
+        incl, omega, Omega,
+        M_planet, M_star,
+    )
+end
+
+function KeplerianOrbit(nt::NamedTuple{(
+        :period, :t_0, :t_p, :duration,
+        :a, :R_planet, :R_star,
+        :rho_star,
+        :RpRs, :aR_star, :b, :ecc, :cos_omega, :sin_omega,
+        :incl, :omega, :Omega,
+        :M_planet, :M_star,
+    )})
+    if nt.period isa Real || nt.a isa Real
+        G = G_nom
+        no_units = true
+    else
+        G = G_unit
+        no_units = false
+    end
```

Comment:
Ok, I can start trying that out. Just checking, but shouldn't the last two Gs be flipped:

```julia
mass_from_kep3(P, a) = mass_from_kep3(P, a, G_nom)
mass_from_kep3(P::AbstractQuantity, a::AbstractQuantity) = mass_from_kep3(P, a, Unitful.G)
```

since `G_nom` is assumed to be unitless here?